crypto/asn1/tasn_new.c

```diff
@@ -142,22 +143,34 @@ int asn1_item_embed_new(ASN1_VALUE **pval, const ASN1_ITEM *it, int embed)
     return 1;
 
  memerr2:
-    ASN1_item_ex_free(pval, it);
+    for (tt = it->templates; i > 0; i--, tt++) {
+        pseqval = asn1_get_field_ptr(pval, tt);
+        asn1_template_free(pseqval, tt);
+    }
+    asn1_do_lock(pval, -1, it);
+ memerr1:
+    if (embed == 0) {
+        OPENSSL_free(*pval);
+        *pval = NULL;
+    }
  memerr:
-    ASN1err(ASN1_F_ASN1_ITEM_EMBED_NEW, ERR_R_MALLOC_FAILURE);
+    ASN1err(ASN1_F_ASN1_ITEM_EMBED_NEW, err);
 #ifndef OPENSSL_NO_CRYPTO_MDEBUG
     OPENSSL_mem_debug_pop();
 #endif
     return 0;
 
  auxerr2:
-    ASN1_item_ex_free(pval, it);
+    err = ASN1_R_AUX_ERROR;
```

Comment:
Yes, I am open for suggestions to find the least-ugly solution here.
Following @FdaSilvaYY 's comment I already lined up the different
labels avoiding the back-jumps.